hera_mc/geo_handling.py

```diff
@@ -19,298 +21,397 @@
-        station_name = to_find.upper()
-    found_it = False
-    if station_name:
-        db = mc.connect_to_mc_db(args)
-        with db.sessionmaker() as session:
-            station_type = session.get_station_type()
-            for a in session.query(geo_location.GeoLocation).filter(geo_location.GeoLocation.station_name == station_name):
-                for key in station_type.keys():
-                    if a.station_name in station_type[key]['Stations']:
-                        this_station = key
-                        break
-                    else:
-                        this_station = 'No station type data.'
-                a.gps2Time()
-                ever_connected = geo_location.is_in_connections(args, a.station_name, '<')
-                active = geo_location.is_in_connections(args, a.station_name, query_date)
-                found_it = True
-                hera_proj = Proj(proj='utm', zone=a.tile, ellps=a.datum, south=True)
-                a.lon, a.lat = hera_proj(a.easting, a.northing, inverse=True)
-                found_location = copy.copy(a)
-                if show_location:
-                    if args.verbosity == 'm' or args.verbosity == 'h':
-                        print('station_name: ', a.station_name)
-                        print('\teasting: ', a.easting)
-                        print('\tnorthing: ', a.northing)
-                        print('\tlon/lat:  ', a.lon, a.lat)
-                        print('\televation: ', a.elevation)
-                        print('\tstation description ({}):  {}'.format(this_station, station_type[this_station]['Description']))
-                        print('\tever connected:  ', ever_connected)
-                        print('\tactive:  ', active)
-                        print('\tcreated:  ', cm_utils._get_displayTime(a.created_date))
-                    elif args.verbosity == 'l':
-                        print(a, this_station)
-    if show_location:
-        if not found_it and args.verbosity == 'm' or args.verbosity == 'h':
-            print(args.locate, ' not found.')
-    return found_location
-
-def get_all_locations(args):
-    db = mc.connect_to_mc_db(args)
-    with db.sessionmaker() as session:
-        stations = session.query(geo_location.GeoLocation).all()
-        connections = session.query(part_connect.Connections).all()
-        stations_new = []
+        Parameters:
+        ------------
+        args:  needed arguments to open database and set date/time
+        station_name:  string name of station
+        active_date:  astropy Time to check if active, ignores if not an instance of Time
+        """
+
+        connected_station = self.session.query(part_connect.Connections).filter(part_connect.Connections.upstream_part == station_name)
+        if connected_station.count() > 0:
+            station_connected = True
+        else:
+            station_connected = False
+        if station_connected and isinstance(active_date,Time):
```

Comment:
what happens if active_date is not a Time object? I notice below (line 272) that you call this method and pass '>' for the active_date, but it looks like that will just get ignored here. I really think you should test that active_date is the type you expect and raise an error if it's not. Silently ignoring it leads to hard to diagnose problems in the future.